Assets/Scripts/ManaSystem.cs

```diff
@@ -29,62 +28,89 @@ public float GetCurrentMana()
 
     /* This method allows the player to gain mana when called elsewhere
      * 
-     * If manaGained would put the player over maxMana:
-     *      Set currentMana to maxMana
-     * Otherwise:
+     * If manaGained is positive:
+     *      Set a temporary variable for the currentMana that is used in the ChangeFillAmount() Lerp function
      *      Set currentMana to the value of currentMana plus manaGained
+     *      Clamp this value to the maxMana so that the player never has more than max mana
+     *      ChangeFillAmount() based on the new mana level
+     *      
+     * Otherwise manaGained must be negative:
+     *      As such, the programer is told via Debug.LogError that they likely intended to use ConsumeMana()
      *      
-     * Update the UI FillAmount to be (new currentMana)/maxMana
      */
-    public void GainMana(float manaGained)
+    public void GainMana(float manaGain)
     {
-        if (currentMana + manaGained >= maxMana)
+        if (manaGain > 0)
         {
-            currentMana = maxMana;
+            float manaBefore = currentMana;
+            currentMana += manaGain;
+            Mathf.Clamp(currentMana, 0, maxMana);
```

Comment:
Mathf.Clamp does not take variables by reference. instead, it returns the new value so you'll have to do current mana = Mathf.Clamp etc.